### PR TITLE
[@mantine/core] Segmented Control: add vertical orientation

### PR DESCRIPTION
**Pull request by hammo92**

### CODE REVIEW

File: src/mantine-core/src/components/SegmentedControl/SegmentedControl.styles.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ interface SegmentedControlStyles {
   transitionDuration: number;
   transitionTimingFunction: string;
   size: MantineSize;
+  vertical: boolean;
```
Comment (rtivital) [inline src/mantine-core/src/components/SegmentedControl/SegmentedControl.styles.ts:13]: This prop should be renamed to `orintation` and have type `'vertical' | 'horizontal'` to be consistent with other components

File: src/mantine-core/src/components/SegmentedControl/SegmentedControl.styles.ts [hunk 2]
```diff
@@ -75,9 +77,9 @@ export default createStyles(
         }`,
 
         '&:not(:first-of-type)': {
-          borderLeft: `1px solid ${
-            theme.colorScheme === 'dark' ? theme.colors.dark[4] : theme.colors.gray[3]
-          }`,
+          border: 'solid',
```
Comment (rtivital) [inline src/mantine-core/src/components/SegmentedControl/SegmentedControl.styles.ts:80]: Replace `border` with `borderStyle` to be more specific

File: src/mantine-core/src/components/SegmentedControl/demos/vertical.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
```
Comment (rtivital) [inline src/mantine-core/src/components/SegmentedControl/demos/vertical.tsx:1]: This demo is redundant, it should be combined with `fullWidth` demo. `fullWidth` demo should be a configurator that has option to change orientation.

**Review by rtivital — CHANGES_REQUESTED** [review]
Thanks for PR, it works great! Please make adjustments to the demos and code to merge this PR.

File: src/mantine-core/src/components/SegmentedControl/demos/vertical.tsx (hunk 3 above)
Comment (hammo92) [reply src/mantine-core/src/components/SegmentedControl/demos/vertical.tsx:1]: Rename this demo in the docs, or keep the header as "Full width"?

Comment (rtivital) [reply src/mantine-core/src/components/SegmentedControl/demos/vertical.tsx:1]: You can leave the name as is, I'll clean up docs later

**Comment by rtivital** [comment]
Thanks!

**Pull request merged**